extensions/investspec/investspec.py

```diff
@@ -0,0 +1,418 @@
+import docutils
+import importlib
+
+INPUT_TYPES_HTML_FILE = 'input_types.html'
+# accepted geometries for a vector will be displayed in this order
+GEOMETRY_ORDER = [
+    'POINT',
+    'MULTIPOINT',
+    'LINESTRING',
+    'MULTILINESTRING',
+    'POLYGON',
+    'MULTIPOLYGON']
+
+
+def format_type_string(arg_type):
+    """Represent an arg type as a user-friendly string.
+
+    Args:
+        arg_type (str): the type to format
+
+    Returns:
+        formatted string that links to a description of the input type
+    """
+    # Represent the type as a string. Some need a more user-friendly name.
+    # we can only use standard docutils features here, so no :ref:
+    # this syntax works to link to a section in a different page, but it
+    # isn't universally supported and depends on knowing the built page name.
+    if arg_type == 'freestyle_string':
+        return f'`text <{INPUT_TYPES_HTML_FILE}#text>`__'
+    elif arg_type == 'option_string':
+        return f'`option <{INPUT_TYPES_HTML_FILE}#option>`__'
+    elif arg_type == 'boolean':
+        return f'`true/false <{INPUT_TYPES_HTML_FILE}#truefalse>`__'
+    else:
+        return f'`{arg_type} <{INPUT_TYPES_HTML_FILE}#{arg_type}>`__'
+
+
+def format_units_string(unit):
```

Comment:
This is a good point. Lets have a call to discuss this because it relates to some largers ARGS_SPEC stuff.

Reply:
@emlys I was just looking at https://pint.readthedocs.io/en/0.10.1/tutorial.html#string-formatting and thinking it would be great to apply the built-in formatting options if possible. But since we need an RST output, maybe that doesn't fully solve the problem?

For the workbench though we could probably use the HTML print options. So maybe all we need then is another `ui_server.py` endpoint to return that markup. If so, maybe it's not so important that the UG and the Workbench share a formatting function?